src/Adapter/Currency/CommandHandler/AddUnofficialCurrencyHandler.php

```diff
@@ -87,7 +87,7 @@ public function handle(AddUnofficialCurrencyCommand $command)
             $entity = $this->currencyDataProvider->getCurrencyByIsoCodeOrCreate($command->getIsoCode()->getValue());
 
             $entity->unofficial = true;
-            $entity->numeric_iso_code = null;
+            $entity->numeric_iso_code = '';
```

Comment:
Mmm, careful with this one I'm not sure any more but I believe the `null` value was expected Maybe the tests won't like it

Reply:
Yep in the behat context `CurrencyFeatureContext::assertCurrencyNumericIsoCode` checks that it IS null, for non official currencies the value `null` is not only allowed but expected

Reply:
Ahd the DB allows it as well, so I guess you need to change the PHPDoc instead to `string|null`

Reply:
The database doesn't allow null values in this field : 

```sql
ALTER TABLE `PREFIX_currency` ADD `numeric_iso_code` varchar(3) NOT NULL DEFAULT '0' AFTER `iso_code`;
```

from `install-dev/upgrade/sql/1.7.6.0.sql` file

Reply:
But in `install-dev/data/db_structure.sql`

```sql
CREATE TABLE `PREFIX_currency` (
  `id_currency` int(10) unsigned NOT NULL auto_increment,
  `name` varchar(64) NOT NULL, /* Deprecated since 1.7.5.0. Use PREFIX_currency_lang.name instead. */
  `iso_code` varchar(3) NOT NULL DEFAULT '0',
  `numeric_iso_code` varchar(3),
```

Bug to come ?!

Reply:
Fixed